lib/rbi/rbs_printer.rb

```diff
@@ -273,19 +273,21 @@ def visit_attr(node)
 
     sig { params(node: RBI::Attr, sig: Sig).void }
     def print_attr_sig(node, sig)
+      ret_type = sig.return_type
+
       type = case node
-      when AttrAccessor, AttrReader
-        parse_type(sig.return_type)
+      when AttrReader, AttrAccessor
+        parse_type(ret_type).rbs_string
       else
-        first_arg = sig.params.first
-        if first_arg
-          parse_type(first_arg.type)
+        arg_type = sig.params.first
+        if arg_type && (ret_type.is_a?(Type::Void) || ret_type == "void")
+          parse_type(arg_type.type).rbs_string
         else
-          Type.untyped
+          parse_type(ret_type).rbs_string
```

Comment:
Not blocking, but I think it would be helpful to have a comment here that in the case of mismatching arg and return types, Sorbet will default to the return type